packages/orchestration/src/util/stream.ts

```diff
@@ -0,0 +1,345 @@
+import { createLogger } from '@sap-cloud-sdk/util';
+import type {
+  ChatDelta,
+  ChoiceLogprobs,
+  CompletionPostResponseStreaming,
+  LlmChoice,
+  LlmChoiceStreaming,
+  LlmModuleResult,
+  LLMModuleResultStreaming,
+  MessageToolCall,
+  ModuleResults,
+  ModuleResultsStreaming,
+  OrchestrationStreamChunkResponse,
+  OrchestrationStreamResponse,
+  ResponseChatMessage,
+  ToolCallChunk
+} from '../index.js';
+
+const logger = createLogger({
+  package: 'orchestration',
+  messageContext: 'stream-util'
+});
+
+/**
+ * @internal
+ */
+export function mergeStreamResponse(
+  chunk: CompletionPostResponseStreaming,
+  response: OrchestrationStreamResponse<OrchestrationStreamChunkResponse>
+): void {
+  const data = response._data;
+  data.request_id = chunk.request_id;
+  data.module_results = mergeModuleResults(
+    data.module_results,
+    chunk.module_results
+  );
+  data.orchestration_result = mergeLlmModule(
+    data.orchestration_result,
+    chunk.orchestration_result
+  );
+}
+
+function mergeModuleResults(
+  existing: ModuleResults | undefined,
+  incoming: ModuleResultsStreaming | undefined
+): ModuleResults {
+  const mergedModuleResults = { ...existing };
+  for (const [moduleName, moduleResult] of Object.entries(incoming || {})) {
+    switch (moduleName) {
+      case 'llm':
+        mergedModuleResults[moduleName] = mergeLlmModule(
+          mergedModuleResults[moduleName],
+          moduleResult
+        );
+        break;
+      case 'output_unmasking':
+        mergedModuleResults[moduleName] = mergeLlmChoices(
+          mergedModuleResults[moduleName],
+          moduleResult
+        );
+        break;
+      default:
+        mergedModuleResults[moduleName] = moduleResult;
+    }
+  }
+  return mergedModuleResults;
+}
+
+function mergeLlmModule(
```

Comment:
```suggestion
function mergeOrchestrationResult(
```
[pp] I think name it after the property name makes it more clear.